### PR TITLE
feat: Added SARIF command "appknox sarif <fileID> --risk-threshold <low,medium etc> --output <filepath>" 

### PR DESCRIPTION
**Pull request by ashujha301**
### Changelog

1. Added SARIF command.
2. Modified appknox/helper/sarif.go, appknox/cmd/sarif.go.
3. Added sarif generator and its test.

---

#### Dependent PRs

-   https://github.com/appknox/appknox-github-action/pull/5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -191,6 +191,35 @@ ID      RISK    CVSS-VECTOR                                   CVSS-BASE  VULNERA
 
 exit status 1
 
+- **Upload a file and do cicheck based on the risk threshold with sarif flag for sarif report generation**
+
+$ appknox upload ~/Downloads/mfva.apk | xargs appknox cicheck --risk-threshold low --sarif
```
Comment (ginilpg) [inline README.md]: Instead of mfva,apk, say "appname.apk"

File: README.md [hunk 2]
```diff
@@ -191,6 +191,35 @@ ID      RISK    CVSS-VECTOR                                   CVSS-BASE  VULNERA
 
 exit status 1
 
+- **Upload a file and do cicheck based on the risk threshold with sarif flag for sarif report generation**
```
Comment (ginilpg) [inline README.md]: use ###  for comments/subheadings. remove hyphen and asterisks

File: README.md [hunk 3]
```diff
@@ -191,6 +191,35 @@ ID      RISK    CVSS-VECTOR                                   CVSS-BASE  VULNERA
...
+2.3 MiB / 2.3 MiB [==========================================================| 00:00 ] 226.28 KiB/s
+Static Scan Progress:  100 % [==========================================================| ]
+SARIF report created successfully at: /file/path/for/sarif/report.sarif
+Found vulnerabilities with risk threshold greater or equal than the provided: Low
+
+ID      RISK    CVSS-VECTOR                                   CVSS-BASE  VULNERABILITY-ID  VULNERABILITY-NAME
+--      ----    -----------                                   ---------  ----------------  ------------------
+671660  High    CVSS:3.0/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:L  7.3        2                 Improper Content Provider Permissions
+671637  High    CVSS:3.0/AV:L/AC:H/PR:N/UI:R/S:C/C:H/I:H/A:H  7.7        3                 Application Debugging
+671635  Low     CVSS:3.0/AV:L/AC:L/PR:H/UI:N/S:U/C:L/I:N/A:N  2.3        10                Unused Permissions
```
Comment (ginilpg) [inline README.md]: Specify only 2-3 lines. No need to mention entire results

**Comment by cosmosgenius** [comment]
lets move the sarif implementation into the appknox folder. The reasoning being sarif can be using via appknox go import... helper are used for cli alone...

Can we can add command which generates sarif given a report id?
cicheck is not needed then for sarif as it can be piped. If flag is need now it can be handled outside the CICheckProcess function along with sarif function.

**Comment by cosmosgenius** [comment]
can you move the sarif implemention helper/sarif.go inside appknox folder. Will be needs tests for the same as well. 

Add two functions:

1. Generate Sarif Given Report ID 
    This will Generate the Sarif given report ID and return the Sarif struct.

2. Generate Sarif file content given a Sarif struct.
    This will Generate a Sarif string content given a Sarif struct.

The helper code will still be needed to do the rest of the things but will use the above two function to generate it.
Tests will be required for the above two function and structs as well. Please take advantage of code coverage generated using make `test_coverage` to verify the test coverage.

**Comment by ashujha301** [comment]
> can you move the sarif implemention helper/sarif.go inside appknox folder. Will be needs tests for the same as well.
> 
> Add two functions:
> 
> 1. Generate Sarif Given Report ID
>    This will Generate the Sarif given report ID and return the Sarif struct.
> 2. Generate Sarif file content given a Sarif struct.
>    This will Generate a Sarif string content given a Sarif struct.
> 
> The helper code will still be needed to do the rest of the things but will use the above two function to generate it. Tests will be required for the above two function and structs as well. Please take advantage of code coverage generated using make `test_coverage` to verify the test coverage.

Is it necassary to use reportID rather than fileID for generating SARIF?

**Review by cosmosgenius — APPROVED** [review]

**Pull request merged**